src/main/java/teammates/ui/controller/StudentCourseJoinAction.java

```diff
@@ -20,12 +22,23 @@
     @Override
     public ActionResult execute() {
         Assumption.assertPostParamNotNull(Const.ParamsNames.REGKEY, regkey);
-        String nextUrl = getNextUrl();
         
         statusToAdmin = "Action Student Clicked Join Link"
                         + (account.googleId == null ? "<br>Email: " + account.email
                                                     : "<br>Google ID: " + account.googleId + "<br>Key: " + regkey);
         
+        if (student == null) {
+            statusToAdmin += "<br>Student course join failed due to being deleted by instructor.";
```

Comment:
Reword this status as well, to `<br>Student course join failed as student does not exist.`